### PR TITLE
qgis3: fix ccache

### PR DESCRIPTION
**Pull request by fjperini**
@nickrobison @luisspuerto solution to the problem if `ccache` is previously installed: `/bin/sh: ccache: command not found`

### CODE REVIEW

**Comment by nickrobison** [comment]
@fjperini When performing a cold build of `qgis3` what's the hit rate in ccache?

Seems like there would be very little benefit to the user, since they would be installing QGIS for the first time and thus the cache wouldn't have any object files to use. But perhaps I'm missing something?

File: Formula/qgis3.rb [hunk 1]
```diff
@@ -73,6 +73,7 @@ def pour_bottle?
   option "with-qspatialite", "Build QSpatialite Qt database driver"
   option "with-api-docs", "Build the API documentation with Doxygen and Graphviz"
   option "with-3d", "Build with 3D Map View panel"
+  option "with-ccache", "Build with CCache"
```
[the remark this answers is not in the document]
Comment (fjperini) [reply Formula/qgis3.rb]: I remove it now or when I upload other changes?

[the remark this answers is not in the document]
Comment (nickrobison) [reply Formula/qgis3.rb]: Please remove it now. That way we can keep all the changes grouped together.

I know it's a hassle, but I'm trying to clean up the code base to reduce duplicated features and general code sprawl. Thanks!

**Comment by fjperini** [comment]
If you consider that it is not so necessary, we can only add: `-DUSE_CCACHE=OFF`

File: Formula/qgis3.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (fjperini) [reply Formula/qgis3.rb]: Done!

**Comment by nickrobison** [comment]
My concern is that a user tries to install QGIS with all the flags enabled and the end result is that we're now support 4 different build combinations:

ninja
make
ccache+make
ccache+ninja

Which seems like a lot, and creates the potential for a lot of things to go wrong. But I can understand the desire to speed up really long, repeated builds.

Looks like there's only one formula in core that has a option for building with ccache. Perhaps a better solution is to simply locally alias the CXX and CC variables on the developer's machine?

**Comment by fjperini** [comment]
I also noticed the same, if it is a good solution. Can you take care of those changes?

**Comment by nickrobison** [comment]
I think my recommended solution is to simply let the developer enable ccache using the `--cc` option in Homebrew, which would not require any changes to the formula. Would that be a good solution?

That would mean only adding the `-DUSE_CCACHE=OFF` option to the `qgis3` formula.

**Comment by fjperini** [comment]
Ok! I will leave only that option.

**Pull request merged**